rpmlint/checks/FileDigestCheck.py

```diff
@@ -120,23 +239,19 @@ def _resolve_links(self, pkg, path):
 
         return pkgfile
 
-    def _calc_digest(self, pkgfile, algorithm):
-        pair = (pkgfile.name, algorithm)
+    def _calc_digest(self, group_type, pkgfile, algorithm):
+        # include the group_type in the cache key, because different groups
+        # can use different Digester types
+        cache_key = (group_type, pkgfile.name, algorithm)
 
-        digest = self.digest_cache.get(pair, None)
+        digest = self.digest_cache.get(cache_key, None)
         if digest is not None:
             return digest
 
-        h = hashlib.new(algorithm)
-        with open(pkgfile.path, 'rb') as fd:
-            while True:
-                chunk = fd.read(4096)
-                if not chunk:
-                    break
-                h.update(chunk)
+        Digester = self.digester_for_group[group_type]
```

Comment:
Likely use `digester` here, similarly to previous `digester = DIGESTERS.get(digester_name, None)`.